native/KotlinIOS/KotlinIOS/ViewController.swift

```diff
@@ -31,35 +32,47 @@ class ViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSo
     @IBAction func departureButton(_ sender: Any) {
         let departureStation: String = stationData[departurePicker.selectedRow(inComponent: 0)]
         let arrivalStation: String = stationData[arrivalPicker.selectedRow(inComponent: 0)]
-//        let urlString: String = "https://www.lner.co.uk/travel-information/travelling-now/live-train-times/depart/\(departureStation)/\(arrivalStation)/#LiveDepResults"
-//        let urlComponents = URLComponents (string: urlString)!
-//        UIApplication.shared.open (urlComponents.url!)
         presenter.onButtonTapped(departureStation: departureStation, arrivalStation: arrivalStation, view: self)
-        
     }
     
     func setDepartureDropdown(stationList: Array<String>) {
         self.departurePicker.delegate = self
         self.departurePicker.dataSource = self
         stationData = stationList
     }
-    
     func setLabel(text: String) {
         label.text = text
     }
-    
     func setArrivalDropdown(stationList: Array<String>) {
         self.arrivalPicker.delegate = self
         self.arrivalPicker.dataSource = self
         stationData = stationList
     }
     
-}
     
-
-
-
-
-
-
-
+    
+    @IBOutlet weak var tableView: UITableView!
+    func setupTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
+        self.tableView.reloadData()
+    }
+    func populateDeparturesTable(departuresList: Array<String>){
+        departuresData = departuresList
+        setupTableView()
```

Comment:
Similar to the picker, rather than setting up the table all over again when you change the data, the table has a method `reloadData` which you can use here. Any setup needed for the page, such as calling `setupTableView` should occur in the `viewDidLoad` method, since you know that this will always be called when the page is created. Knowing that this method is called during the creation of the page is reliant on the fact you know how the Presenter is implemented, and generally, we should avoid relying that. Does that make sense?